docs/user-guide/deployments.md

```diff
@@ -556,6 +696,16 @@ the rolling update starts, such that the total number of old and new Pods do not
 the new Replica Set can be scaled up further, ensuring that the total number of Pods running
 at any time during the update is at most 130% of desired Pods.
 
+### Progress Deadline Seconds
+
+`.spec.progressDeadlineSeconds` is an optional field that specifies the number of seconds you want
+to wait for your Deployment to progress before the system reports back that the Deployment has
+[failed progressing](#failed-deployment) - surfaced as a condition with `Type=Progressing`, `Status=False`.
+and `Reason=ProgressDeadlineExceeded` in the status of the resource. The deployment controller will keep
+retrying the Deployment. In the future, once automatic rollback will be implemented, the deployment
+controller will rollback a Deployment as soon as it observes such a condition. If specified, this field
+needs to be greater than `.spec.minReadySeconds`.
```

Comment:
Is this line related to the previous line about auto rollback? If not, separate them (move auto rollback to the end); if so, make it clear that they're relevant